client/allocrunner/taskrunner/task_runner_test.go

```diff
@@ -1108,6 +1108,16 @@ func TestTaskRunner_NoShutdownDelay(t *testing.T) {
 
 	err := <-killed
 	require.NoError(t, err, "killing task returned unexpected error")
+
+	// Check that we only emit the expected events.
+	hasEvent := false
+	for _, ev := range tr.state.Events {
+		require.NotEqual(t, structs.TaskWaitingShuttingDownDelay, ev.Type)
+		if ev.Type == structs.TaskSkippingShutdownDelay {
+			hasEvent = true
+		}
+	}
+	require.True(t, hasEvent)
```

Comment:
Not a blocker, but I think this kind of loop is what [`must.Contains`](https://pkg.go.dev/github.com/shoenig/test#Contains)  was designed for?

Reply:
I couldn't quite make that work. The best I was able to do was something like this:

```go
want := &structs.TaskEvent{Type: structs.TaskSkippingShutdownDelay}
must.SliceContainsFunc(t, tr.state.Events, want, func(a, b *structs.TaskEvent) bool {
	return a.Type == b.Type
})
```

But `SliceNotContainsFunc`, which I would have to use to check for the absence of  `TaskWaitingShuttingDownDelay` doesn't exist yet, so I left the initial implementation but replacing `require` with `must`.

Reply:
> SliceNotContainsFunc

probably should exist, created https://github.com/shoenig/test/issues/110

Reply:
Awesome, thanks!

I was thinking about the ergonomics of this use case, and I think it could be helpful to have something like these:

```go
s := []string{"hit", "miss"}

must.SliceAtLeastOne(s, func(i string) bool {
  return i == "hit"
}) // success

must.SliceAll(s, func(i string) bool {
  return i == "hit"
}) // fail

must.SliceNone(s, func(i string) bool {
  return i == "none"
}) // success
```